### ISSUE TITLE
Add readable/writeable/ready methods to tokio::io::File

### ISSUE THREAD

**Issue by cmleinz** [tokio-rs/tokio#5698]
It would be nice to have these methods along with the `try_*` methods which are available for network I/O be available for tokio File handles. I have a `std::sync::Mutex` containing a buffer which I would like to read into. With a TCP handle I can await on `readable` and only once bytes are available, take the lock, and `try_read_buf` into the underlaying buffer. Currently there isn't a good way of doing this when reading a File, perhaps that's for good reason, but if not it would be very helpful. Thanks!

**Comment by Darksonn** [comment]
Unfortunately, this is not really possible due to technical reasons. The file type offloads its IO to the `spawn_blocking` thread pool, which means that its implementation is completion based rather than readiness based. Making it pretend to have a readiness based API is rather awkward.

You may be able to do what you want by wrapping the file in a `BufReader` and using the `fill_buf` method.

Thank you for the suggestion.
